File: 2006-find-the-student-that-will-replace-the-chalk/2006-find-the-student-that-will-replace-the-chalk.go
func chalkReplacer(chalk []int, k int) int {
    can := 0
    for _, num := range chalk {
        can += num
    }
    k %= can
    for i := 0; i < len(chalk); i++ {
        if chalk[i] > k {
            return i
        }
        k -= chalk[i]
    }
    return -1
}

File: 2006-find-the-student-that-will-replace-the-chalk/2006-find-the-student-that-will-replace-the-chalk.py
class Solution(object):
    def chalkReplacer(self, chalk, k):
        """
        :type chalk: List[int]
        :type k: int
        :rtype: int
        """
        remaining_after_com_rounds = k % sum(chalk)

        for i in range(len(chalk)):
            if remaining_after_com_rounds < chalk[i]:
                return i
            
            remaining_after_com_rounds -= chalk[i]
        
        return -1 #Safeguard, this will never execute